### PR TITLE
Increase month when day out of limit in the dxDateBox with masks (T710987)

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

**Review by KsyushaM — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 1]
```diff
@@ -121,14 +121,18 @@ if(devices.real().deviceType === "desktop") {
         });
 
         test("Day", (assert) => {
-            checkAndRemoveLimits(this.parts[4], { min: 1, max: 29 }, assert);
+            checkAndRemoveLimits(this.parts[4], { min: 1, max: 31 }, assert);
+
+            let date = new Date(2012, 1, 4, 15, 6);
+            this.parts[4].setter(date, 3);
+            assert.equal(date.getDate(), 3, "setter sets day");
```
Comment (churkin) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: strictEqual

File: js/ui/date_box/ui.date_box.mask.parts.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const monthSetter = (date, value) => {
 
     date.setMonth(newValue - 1, 1);
 
-    let dayLimits = getLimits("d", date),
+    let dayLimits = getLimits("dM", date),
```
Comment (churkin) [inline js/ui/date_box/ui.date_box.mask.parts.js]:
```javascript
const { min, max } = getLimits("dM", date);
const newDay = fitIntoRange(day, min, max);
```

**Review by churkin — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 3]
```diff
@@ -139,7 +143,7 @@ if(devices.real().deviceType === "desktop") {
 
             let date = new Date(2012, 1, 4, 15, 6);
             this.parts[6].setter(date, 15);
-            assert.equal(date.getFullYear(), 2015, "setter sets AM");
+            assert.equal(date.getFullYear(), 2015, "setter sets year");
```
Comment (churkin) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: strictEqual

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 4]
```diff
@@ -655,7 +659,11 @@ if(devices.real().deviceType === "desktop") {
             assert.equal(this.$input.val(), "Feb, 03", "select on typing");
 
             this.keyboard.type("1");
-            assert.equal(this.$input.val(), "Feb, 01", "don't accept out-of-limit values");
+            assert.equal(this.$input.val(), "Mar, 31", "current month overflow should increase month");
```
Comment (churkin) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: strictEqual

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 5]
```diff
@@ -655,7 +659,11 @@ if(devices.real().deviceType === "desktop") {
...
+
+            this.keyboard.press("enter");
+            this.keyboard.type("35");
+            assert.equal(this.$input.val(), "Mar, 05", "out-of-limit values should clear search value");
```
Comment (churkin) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: strictEqual

**Review by churkin — CHANGES_REQUESTED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**